provision.sh

```diff
@@ -239,24 +256,35 @@ install_misc_tools() {
 
 install_language_servers() {
   echo ">>> Installing language servers"
-  snap install bash-language-server --classic
-  snap install typescript-language-server
+  #snap install bash-language-server --classic
+  #snap install typescript-language-server
+  apt-get install -y npm node-read-package-json
+  npm install -g typescript-language-server typescript
+  npm i -g bash-language-server
 }
 
 install_github_cli() {
   echo ">>> Installing Github CLI"
-  wget -qO- https://cli.github.com/packages/githubcli-archive-keyring.gpg >/usr/share/keyrings/githubcli-archive-keyring.gpg
-  echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/githubcli-archive-keyring.gpg] https://cli.github.com/packages stable main" >/etc/apt/sources.list.d/github-cli.list
+  curl -fsSL https://cli.github.com/packages/githubcli-archive-keyring.gpg | sudo dd of=/usr/share/keyrings/githubcli-archive-keyring.gpg \
+    && sudo chmod go+r /usr/share/keyrings/githubcli-archive-keyring.gpg \
+    && echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/githubcli-archive-keyring.gpg] https://cli.github.com/packages stable main" | sudo tee /etc/apt/sources.list.d/github-cli.list > /dev/null 
```

Comment:
Using `tee` in combination with `> /dev/null` is effectively the same as just redirecting to the file, but the latter is more concise.

Reply:
cleaned up all sudos

Reply:
tee is now only used to display the added repository in all cases